---
layout: page
title: Instructors
description: CPANG18 Course instructors description
---

## Tobias Marschall
**Affiliation:** Center for Bioinformatics at Saarland University and Max Planck Institute for Informatics

<img src="https://www.mpi-inf.mpg.de/~marschal/marschal.jpg" height="220" width="170" align="left" style="margin-right: 3%; margin-bottom: 0.3em;">

[Tobias Marschall ](http://www.zbi.uni-saarland.de/en/junior-groups/algorithms-computational-genomics.html) is an assistant professor at the Center for Bioinformatics at Saarland University and affiliated with the Max Planck Institute for Informatics as a senior researcher. He is heading the Algorithms for Computational Genomics group. 

His research targets algorithmic and statistical challenges arising from present-day genomics technologies. A particular focus is on population-scale sequencing efforts, on structural variation discovery, on algorithms and data structures for pangenomes and on haplotyping in various contexts (from humans to viruses). He has co-organized two workshops on computational pangenomics, one at the [Lorentz Center in Leiden, The Netherlands,](https://lorentzcenter.nl/lc/web/2015/698/info.php3?wsid=698&venue=Oort), which has lead to a [comprehensive review paper](https://doi.org/10.1093/bib/bbw089), and one at [ECCB 2016](https://www.ebi.ac.uk/eccb/2016/programme/workshops/w9/index.html)

Tobias received his undergraduate education at Bielefeld University (Germany), where he studied computer science and physics. He obtained his PhD in bioinformatics from TU Dortmund. After that, he moved to Centrum Wiskunde & Informatica (Amsterdam, The Netherlands), the Dutch national institute for mathematics and computer science, as a postdoctoral researcher. During his time as a postdoc, he also was a long-term participant of the semester program on Mathematical and Computational Approaches in High-Throughput Genomics held at the Institute for Pure and Applied Mathematics (IPAM) at University of California Los Angeles (UCLA).

---

### Erik Garrison
**Affiliation:** Wellcome Trust Sanger Institute and Cambridge University

<img src="https://raw.githubusercontent.com/Pfern/PANGenomics/master/images/erik-garrison.jpg" height="220" width="170" align="left" style="margin-right: 3%; margin-bottom: 0.3em;">

[Erik Garrison](http://hypervolu.me/~erik/erik_garrison.html) is a PhD student at the Wellcome Trust Sanger Institute and Cambridge University. His ongoing doctoral research has focused on the development of a software toolkit for practical pangenomics: [vg](https://github.com/vgteam/vg). 

He has seven years of experience in genomics, where he has worked on the development of sequencing systems, participated in large scale sequencing projects such as the [1000 Genomes Project](http://www.internationalgenome.org/), and authored popular bioinformatics software such as the [freebayes](https://github.com/ekg/freebayes) variant caller. 

Raised in Kentucky, Erik obtained an undergraduate degree in the social sciences from Harvard University. After graduation he worked at Harvard Medical School, One Laptop Per Child, and Boston College before beginning his studies at the Sanger Institute.

<br/>

## Course Assistant

### Jordan Eizenga
**Affiliation:** University of California Santa Cruz

<img src="https://cgl.genomics.ucsc.edu/wp-content/uploads/2016/07/CGL_Team_Pics_Jordan_Einzeinga.png" height="220" width="170" align="left" style="margin-right: 3%; margin-bottom: 0.3em;">

[Jordan Eizenga](https://github.com/jeizenga) is a PhD student in the Computational Genomics Lab at the University of California Santa Cruz. His advisors are David Haussler and Benedict Paten. Jordan was originally trained as a mathematician, but made the switch to bioinformatics in 2015. 

Now his research focuses on using graph data structures in genomics. He has published on both theoretical and practical aspects of this problem, and he has been heavily involved in the development of the [vg](https://github.com/vgteam/vg) project.

<br/>
<br/>
<br/>

### Back

Back to [main page](../index.md).
